goasupport/conditional_request/generator.go

```diff
@@ -450,22 +458,34 @@ func (ctx *{{$resp.Name}}) getIfNoneMatch() *string {
 }`
 
 	matchesETag = `
-// matchesETag returns 'true' the given 'etag' argument matches with the context's 'IfNoneMatch' value.
-func matchesETag(ctx ConditionalRequestContext, etag string) bool {
-	if ctx.getIfNoneMatch() != nil && *ctx.getIfNoneMatch() == etag {
-		return true
+// matchesETag compares the given 'etag' argument matches with the context's 'IfNoneMatch' value.
+func matchesETag(ctx ConditionalRequestContext, etag string) (bool, bool) {
+	if ctx.getIfNoneMatch() != nil {
+		if *ctx.getIfNoneMatch() == etag {
+			// 'If-None-Match' field was found and matched with given 'etag' argument
+			return true, true
+		}
+		// 'If-None-Match' field was found and but did not match with given 'etag' argument
+		return true, false
 	}
-	return false
+	// 'If-None-Match' field was not found
+	return false, false
 }`
 
 	modifiedSince = `
-// modifiedSince returns 'true' if the given context's 'IfModifiedSince' value is before the given 'lastModified' argument
-func modifiedSince(ctx ConditionalRequestContext, lastModified time.Time) bool {
+// modifiedSince compares the given context's 'IfModifiedSince' value is before the given 'lastModified' argument
+func modifiedSince(ctx ConditionalRequestContext, lastModified time.Time) (bool, bool) {
```

Comment:
Would be more clear if there were comments what each of that bool return objects means.